### PR TITLE
DIRECTOR: LINGO: Implement MENUREF Datum type and STUB for Menu, Menus, MenuItem and MenuItems

### PR DESCRIPTION
**Pull request by r41k0u**
This change:

- Implements the MENUREF Datum type to refere to menus and menuItems
- Implements the missing getter and setter functions for the STUB.
- Cleans up the old way of findMenuItem() (It would have required 4 overloads)
With these changes, all the workshop movies concerning the menus and menuItems are working.

### CODE REVIEW

**Comment by rvanlaar** [comment]
Good work. I'ld like to see two changes for me to merge the PR. 
Please clean up the commits:
- remove the first commit: it has changes that are made irrelevant by the second commit
- explain the why and how on the second commit

**Comment by rvanlaar** [comment]
Hi @r41k0u 
Yes, I see the first part, thank you. But you seem to have missed the second part.

**Comment by bluegr** [comment]
@r41k0u Please change your commit message according to our guidelines (you removed the DIRECTOR: part)

File: engines/director/lingo/lingo.cpp [hunk 1]
```diff
@@ -947,6 +954,8 @@ Common::String Datum::asString(bool printonly) const {
 
 		s += ")";
 		break;
+	case MENUREF:
+		break;
```
Comment (sev-) [inline engines/director/lingo/lingo.cpp:958]: This is not quite good and needs to be addressed after the merge. We use this for printing the stack contents

**Comment by sev-** [comment]
Thank you

**Pull request merged**